src/test/mod.rs

```diff
@@ -0,0 +1,50 @@
+pub mod jaegar;
+
+use opentelemetry_api::trace::{SpanId, TraceId};
+use rand::Rng;
+
+pub trait Response {
+    fn text(&self) -> String;
+}
+
+pub struct TracedResponse<T> {
+    response: T,
+    pub trace_id: TraceId,
+    pub span_id: SpanId,
+}
+
+impl<T> TracedResponse<T>
+where
+    T: Response,
+{
+    pub fn new(response: T, trace_id: TraceId, span_id: SpanId) -> Self {
+        Self {
+            response,
+            trace_id,
+            span_id,
+        }
+    }
+
+    #[allow(dead_code)]
```

Comment:
I don't think we need `allow(dead_code)` here since this is a library crate and `TracedResponse` is exported.